cromshell

```diff
@@ -282,7 +282,44 @@ function checkPrerequisites
   return 0
 }
 
-function checkForComplexHelpArgument() 
+function assertRequiredFileIsNonEmpty()
+{
+
+  local fileName=$1
+  local fileDescription=$2
+  if [[ -z ${fileName} ]]
+	then
+		error "Error: required file ${fileDescription} was not specified."
+		exit 6
+	else
+		if [[ -f ${fileName} ]]
+		then
+			if [[ -s ${fileName} ]]
+			then
+				return 0
+			else
+				error "Error: file ${fileName} is an empty file."
+				exit 6
+			fi
+		else
+			error "Error: file ${fileName} does not exist."
```

Comment:
Similarly, I would rewrite as `"ERROR: ${fileDescription} file does not exist: ${fileName}"`